src/hooks/utility/view.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::ffi::CStr;
+
+use anyhow::Result;
+use pg_sys::{get_relname_relid, RangeVarGetCreationNamespace};
+
+use pgrx::*;
+
+use crate::{duckdb::connection::execute, hooks::query::is_duckdb_query};
+
+use super::set_search_path_by_pg;
+
+pub fn view_query(query_string: &core::ffi::CStr, stmt: *mut pg_sys::ViewStmt) -> Result<bool> {
+    let query = unsafe { (*stmt).query as *mut pg_sys::SelectStmt };
+    let from_clause = unsafe { (*query).fromClause };
+
+    if analyze_from_clause(from_clause)? {
```

Comment:
I think we could further simplify the code here. 

 See the Postgres source code.
``` c
ObjectAddress
DefineView(ViewStmt *stmt, const char *queryString,
		   int stmt_location, int stmt_len)
{
	RawStmt    *rawstmt;
	Query	   *viewParse;
	RangeVar   *view;
	ListCell   *cell;
	bool		check_option;
	ObjectAddress address;

	/*
	 * Run parse analysis to convert the raw parse tree to a Query.  Note this
	 * also acquires sufficient locks on the source table(s).
	 */
	rawstmt = makeNode(RawStmt);
	rawstmt->stmt = stmt->query;
	rawstmt->stmt_location = stmt_location;
	rawstmt->stmt_len = stmt_len;

	viewParse = parse_analyze_fixedparams(rawstmt, queryString, NULL, 0, NULL);

```
Here it convert source query to a Query struct which with a field named rtable. 
`rtable` means relations related in this query. So we could just check it. 

Actually `Prepare`  `Create View` and executor hook are very similar.  Check https://github.com/paradedb/pg_analytics/blob/d58443333636a3ce81ac918547618ec0d98185b3/src/hooks/utility/prepare.rs#L99

Reply:
`viewParse = parse_analyze_fixedparams(rawstmt, queryString, NULL, 0, NULL);` here Postgres use function `parse_analyze_fixedparams` which convert a query stirng to Query struct. But it doesn't preform `rewrite`.

There is a situation where nested views occur, for example, when a new view is based on a previously created view. In this case, the rtable will contain the view instead of the underlying relation of the view. This means our checks will be ineffective. To address this issue, we can use `pg_analyze_and_rewrite_fixedparams` (which will rewrite the view into the underlying relation during the rewrite phase). 

The above are some of my speculations. We can give it a try. 👍  :)

Reply:
Note that `pg_analyze_and_rewrite_fixedparams` converts it to a list of `PlannedStmt`, which still contains a field named `rtable`.

Reply:
Thanks for letting me learn something new!